### PR TITLE
Make websockets work in a worker scope

### PR DESCRIPTION
**Pull request by Manishearth**
r? @jdm

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9521)

### CODE REVIEW

**Comment by Manishearth** [comment]
(the failure was tidy; this still passes wpt)

**Comment by Ms2ger** [comment]
Using a `.worker.js` file would be shorter. Also, despite what I said yesterday, I'd like a test for the origin attribute too.

**Comment by jdm** [comment]
@Ms2ger What test is missing from #9367? There's an explicit location.origin test, and lots of test result updates too.

File: tests/wpt/metadata/MANIFEST.json [hunk 1]
```diff
@@ -33717,7 +33717,22 @@
   },
   "local_changes": {
     "deleted": [],
-    "items": {},
+    "items": {
+      "testharness": {
+        "websockets/Send-data-worker.htm": [
+          {
+            "path": "websockets/Send-data-worker.htm",
```
Comment (Ms2ger) [inline tests/wpt/metadata/MANIFEST.json]: I don't think this should be here.

r+ assuming you file a followup about checking the Origin header in workers.

**Comment by Manishearth** [comment]
@bors-servo r=Ms2ger

followup @ https://github.com/servo/servo/issues/9535

**Comment by Manishearth** [comment]
@bors-servo retry

 - #9063

**Comment by Manishearth** [comment]
@bors-servo retry

 - #7791

**Pull request merged**